### DISCUSSION TITLE
Export PnPConnection for later use

### DISCUSSION THREAD

**Discussion by kmbdsz** [pnp/powershell#418] [Q&A] [answered]
Hey guys,
I am trying to export PnPConnection-Obect for later use (connect onece - and than import the connection from HDD - becuase this script should be able to run as a scheduled task - so weblogin is not possible). 
Steps I already tried:
1) Export-Clixml
2) Import-Clixml

After importing the Object, it seemed pretty good - but at a second look the imported object is from another type.
Exported-Objecttype: PnP.PowerShell.Commands.Base.PnPConnection
Imported-Objecttype: Deserialized.PnP.PowerShell.Commands.Base.PnPConnection

With this in mind - I tried to fire Get-PnPList -Connection <imported connection> and dealing with this issue:
"PnP.PowerShell.Commands.Base.PnPConnection" from Typ "Deserialized.PnP.PowerShell.Commands.Base.PnPConnection" can't be converted into Typ "PnP.PowerShell.Commands.Base.PnPConnection"

Do you have any idea in mind? Please help me out.

**Comment by erwinvanhunen** [comment, marked as answer]
The connection contains relevant information for the moment of time it's been created. While you can reuse within the same session, it will not work when you span PowerShell sessions. The reason for this is that we have a very complex hierarchy and dependencies to other libraries in that object. If you want to run a script as a scheduled task, the only way forward would be to create an Azure AD App registration and run your script with -ClientId and -Certificate.
